src/components/Image/types.ts

```diff
@@ -0,0 +1,6 @@
+type ImageProps = {
+    /** Event called with image dimensions when image is loaded */
+    onLoad?: (event: {nativeEvent: {width: number; height: number}}) => void;
+};
```

Comment:
ImageProps are missing some props, you can copy them from [here](https://github.com/Expensify/App/pull/31296/files#diff-43e5dc455bc7b8cff48d2100e061641fd4029c6b2f9c7e2a39a4a0afb79e9e71)

Reply:
@blazejkustra

Thank you for pointing out the `ImageProps`. I've reviewed the reference from [here](https://github.com/Expensify/App/pull/31296/files#diff-43e5dc455bc7b8cff48d2100e061641fd4029c6b2f9c7e2a39a4a0afb79e9e71) and appreciate your suggestion. However, I have some reservations about incorporating these types into my current Pull Request, as they seem tangential to the specific change I'm introducing:

1. **History of Changes:** The file you referenced has undergone several revisions. Directly copying it might inadvertently overwrite its existing history, leading to potential misattribution of changes.
2. **Irrelevant Props:** Some of the props, such as `isAuthTokenRequired`, are not relevant to the `BaseImage` component. Including them could introduce unnecessary complexity or errors.
3. **Platform Specificity:** The `BaseImage` component is tailored for specific platforms. Using the full `ImageProps` could lead to compatibility issues between Expo and React Native Web image, given their differences.

Given these concerns, I propose two alternatives:
- **Option A:** We could remove the `types.ts` file from this Pull Request and defer its introduction to the other PR, maintaining clarity and separation of concerns.
- **Option B:** Alternatively, we could rename the type to `BaseImageProps` and limit its scope to only include essential props like `onLoad`, ensuring compatibility and relevance to the current changes.

I believe either approach would maintain the integrity and focus of the Pull Request while addressing the need for proper typing. I'm open to further discussion on this matter.

Reply:
@kidroca Thank you for your perspective on this one! And I agree with you. Both options sounds good to me, although I'm more inclined to option B.

Reply:
I've pushed an update that implements option B